build/Install-WindowsSdkISO.ps1

```diff
@@ -1,121 +1,217 @@
+
+    if ($isoDrive)
+    {
+        Write-Verbose "$isoPath dismounted"
+        Dismount-DiskImage -ImagePath $isoPath | Out-Null
+    }
 }
 
-function Mount-ISO {
-param (
-  [string] $isoPath
-)
-  # Check if image is already mounted
-  $isoDrive = (Get-DiskImage -ImagePath $isoPath | Get-Volume).DriveLetter
+function Disable-StrongName
+{
+    param ([string] $publicKeyToken = "*")
 
-  if (!$isoDrive) {
-    Mount-DiskImage -ImagePath $isoPath -StorageType ISO
-  }
+    reg ADD "HKLM\SOFTWARE\Microsoft\StrongName\Verification\*,$publicKeyToken" /f | Out-Null
+    if ($env:PROCESSOR_ARCHITECTURE -eq "AMD64")
+    {
+        reg ADD "HKLM\SOFTWARE\Wow6432Node\Microsoft\StrongName\Verification\*,$publicKeyToken" /f | Out-Null
+    }
+}
 
-  $isoVolume = (Get-DiskImage -ImagePath $isoPath | Get-Volume)
-  $isoDrive = $isoVolume.DriveLetter + ":"
+function Test-Admin
+{
+    $identity = [Security.Principal.WindowsIdentity]::GetCurrent()
+    $principal = New-Object Security.Principal.WindowsPrincipal $identity
+    $principal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
+}
 
-  Write-Host "$isoPath mounted to $isoDrive"
+# Static(ish) link for Windows SDK
+# Note: there is a delay from Windows SDK announcements to availability via the static link
+$uri = "https://go.microsoft.com/fwlink/?prd=11966&pver=1.0&plcid=0x409&clcid=0x409&ar=Flight&sar=Sdsurl&o1=$buildNumber"
 
-  return $isoDrive
+if ($env:TEMP -eq $null)
+{
+  $env:TEMP = Join-Path $env:SystemDrive 'temp'
 }
 
-function Dismount-ISO {
-param (
-  [string] $isoPath
-)
-  $isoDrive = (Get-DiskImage -ImagePath $isoPath | Get-Volume).DriveLetter 
-  if ($isoDrive) {
-    Dismount-DiskImage -ImagePath $isoPath
-  }
+$winsdkTempDir = Join-Path $env:TEMP "WindowsSDK"
+
+if (![System.IO.Directory]::Exists($winsdkTempDir))
+{
+  [void][System.IO.Directory]::CreateDirectory($winsdkTempDir)
 }
 
-Write-Output "Getting WinSDK from $uri"
+$file = "winsdk_$buildNumber.iso"
+
+Write-Verbose "Getting WinSDK from $uri"
 $downloadFile = Download-File $winsdkTempDir $uri $file
 
 # TODO Check if zip, exe, iso, etc.
-Write-Output "Mounting ISO $file..."
-$isoDrive = Mount-ISO $downloadFile
 
 
-Write-Host "Installing WinSDK"
-$setupPath = Join-Path $isoDrive "WinSDKSetup.exe"
-Start-Process -Wait $setupPath "/features OptionId.UWPCpp /q"
+try
+{
+    Write-Host -NoNewline "Mounting ISO $file..."
+    Mount-ISO $downloadFile
+    Write-Host "Done"
+
+    $isoDrive = Get-ISODriveLetter $downloadFile
+
+    if (Test-Path $isoDrive)
+    {
+        Write-Host -NoNewLine "Installing WinSDK..."
 
-Dismount-ISO $downloadFile
\ No newline at end of file
+        $setupPath = Join-Path "$isoDrive" "WinSDKSetup.exe"
+        Start-Process -Wait $setupPath "/features OptionId.UWPCpp /q"
+        Write-Host "Done"
+    }
+    else
+    {
+        throw "Could not find mounted ISO at ${isoDrive}"
+    }
+
+    Write-Host -NoNewline "Disabling StrongName for Windows SDK..."
+    if (Test-Admin)
```

Comment:
Realistically, the whole script needs `Test-Admin`. I don't think you can install the SDK without it either.

Reply:
If you are not running as admin, the MSI system detects that and prompts for elevation. If it were non-interactive, the elevation prompt fails with an access denied error.

Reply:
the REG command isn't that smart though. To make this more robust, we should check first to see if we even need to download an SDK (e.g. `Test-WindowsSDK`), and test the registry (`Test-StrongNameDisable`) and only perform those steps as required.

Reply:
I would suggest that `Test-Admin` should be first, and bail out if it fails. Then it could disable the strong name in case the installer ever needs to run anything managed (that could make the IPoverUSB service actually work.)